src/Nethermind/Nethermind.TxPool/Collections/SortedPool.cs

```diff
@@ -411,8 +411,10 @@ public void UpdatePool(Func<TGroupKey, IReadOnlySortedSet<TValue>, IEnumerable<(
         {
             foreach ((TGroupKey groupKey, EnhancedSortedSet<TValue> bucket) in _buckets)
             {
-                changingElements(groupKey, bucket);
-                UpdateGroup(groupKey, bucket, changingElements);
+                if (bucket.Count > 0)
```

Comment:
We shouldn't have empty buckets - in `TryRemove` we are removing whole bucket if last element was just removed